src/coordsim/metrics/metrics.py

```diff
@@ -135,9 +136,8 @@ def add_path_delay(delay):
     metrics['total_path_delay'] += delay
 
     # calc path delay per run; average over num generated flows in run
+    metrics['run_num_path_delays'] += 1
     metrics['run_total_path_delay'] += delay
-    if metrics['run_processed_flows'] > 0:
```

Comment:
I'd simply adjust the if to `if metrics['run_generated_flows'] > 0:` rather than moving the code down.